src/renderers/testing/ReactTestRenderer.js

```diff
@@ -76,7 +81,9 @@ class ReactTestComponent {
     transaction: ReactTestReconcileTransaction,
     context: Object,
   ) {
+    var options = transaction.getTestOptions();
     this._currentElement = nextElement;
+    this._hostNode = options.createNodeMock(nextElement);
```

Comment:
@gaearon not sure how to address this, but in `receiveComponent` the transaction [is not pulled from the pool with the test options](https://github.com/Aweary/react/blob/fe450e84ba0940421ffa6d3ec9ecc50a72494a01/src/renderers/testing/ReactTestMount.js#L110) like it is [during the initial mount](https://github.com/Aweary/react/blob/fe450e84ba0940421ffa6d3ec9ecc50a72494a01/src/renderers/testing/ReactTestMount.js#L82)

Injection makes this kind of a hard case, so I'm wondering what you think we can do here.